src/KeyringController.ts

```diff
@@ -310,17 +365,19 @@ class KeyringController extends EventEmitter {
    * Calls the `addAccounts` method on the given keyring,
    * and then saves those changes.
    *
-   * @param {Keyring} selectedKeyring - The currently selected keyring.
-   * @returns {Promise<object>} A Promise that resolves to the state.
+   * @param selectedKeyring - The currently selected keyring.
+   * @returns A Promise that resolves to the state.
    */
-  async addNewAccount(selectedKeyring) {
+  async addNewAccount(
+    selectedKeyring: Keyring<Json>,
+  ): Promise<KeyringControllerState> {
     const accounts = await selectedKeyring.addAccounts(1);
-    accounts.forEach((hexAccount) => {
+    accounts.forEach((hexAccount: string) => {
```

Comment:
Why is the type necessary here? TypeScript should be able to infer this.

Reply:
[6eff4a1](https://github.com/MetaMask/KeyringController/pull/202/commits/6eff4a1b4d09b68cc4ec3fc3b4d0ee378f153f90)